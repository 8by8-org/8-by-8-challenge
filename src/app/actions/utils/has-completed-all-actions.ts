import { UserType } from '@/model/enums/user-type';
import type { User } from '@/model/types/user';

export function hasCompletedAllActions(user: User | null) {
  return (
    user &&
    user.type === UserType.Hybrid &&
    user.completedActions.electionReminders &&
    user.completedActions.registerToVote
  );
}
