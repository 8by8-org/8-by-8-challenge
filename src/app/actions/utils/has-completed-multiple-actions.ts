import { UserType } from '@/model/enums/user-type';
import type { User } from '@/model/types/user';

export function hasCompletedMultipleActions(user: User | null) {
  let actionsCompleted = 0;

  actionsCompleted += Number(!!user?.completedActions.electionReminders);
  actionsCompleted += Number(!!user?.completedActions.registerToVote);
  actionsCompleted += Number(user?.type === UserType.Hybrid);

  return actionsCompleted > 1;
}
