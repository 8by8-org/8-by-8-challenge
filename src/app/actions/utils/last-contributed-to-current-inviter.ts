import type { ChallengerData } from '@/model/types/challenger-data';

export function lastContributedToCurrentInviter(
  contributedTo: ChallengerData[] = [],
  invitedBy: ChallengerData,
): boolean {
  return (
    !!contributedTo.length &&
    contributedTo.at(-1)?.challengerInviteCode ===
      invitedBy.challengerInviteCode
  );
}
