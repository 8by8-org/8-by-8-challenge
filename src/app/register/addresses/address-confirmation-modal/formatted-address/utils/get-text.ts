import { AddressComponent } from '@/model/types/addresses/address-component';

export function getText(addressComponent: AddressComponent | string) {
  return typeof addressComponent === 'string' ? addressComponent : (
      addressComponent.value
    );
}
