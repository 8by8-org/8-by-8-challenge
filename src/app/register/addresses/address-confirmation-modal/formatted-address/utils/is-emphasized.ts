import { AddressComponent } from '../../../../../../model/types/addresses/address-component';

export function isEmphasized(addressComponent: AddressComponent | string) {
  return typeof addressComponent === 'object' && addressComponent.hasIssue;
}
