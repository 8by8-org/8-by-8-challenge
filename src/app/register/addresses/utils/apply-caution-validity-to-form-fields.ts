import { Validity } from 'fully-formed';
import { AddressesForm } from '../addresses-form';
import { AddressErrorTypes } from '@/model/types/addresses/address-error-types';
import type { AddressErrors } from '@/model/types/addresses/address-errors';
import type { AddressComponent } from '@/model/types/addresses/address-component';

export function applyCautionValidityToFormFields(
  form: InstanceType<typeof AddressesForm>,
  errors: AddressErrors[],
) {
  errors.forEach(error => {
    if (error.type === AddressErrorTypes.MissingSubpremise) {
      form.fields[error.form].fields.streetLine2.setValidityAndMessages(
        Validity.Caution,
        [],
      );
    } else if (error.type === AddressErrorTypes.ReviewRecommendedAddress) {
      for (const entry of Object.entries(error.enteredAddress)) {
        const [fieldName, addressComponent] = entry as [
          keyof (typeof form.fields)[(typeof error)['form']]['fields'],
          AddressComponent,
        ];

        if (addressComponent.hasIssue) {
          form.fields[error.form].fields[fieldName].setValidityAndMessages(
            Validity.Caution,
            [],
          );
        }
      }
    } else if (error.type === AddressErrorTypes.UnconfirmedComponents) {
      for (const entry of Object.entries(error.unconfirmedAddressComponents)) {
        const [fieldName, addressComponent] = entry as [
          keyof (typeof form.fields)[(typeof error)['form']]['fields'],
          AddressComponent,
        ];

        if (addressComponent.hasIssue) {
          form.fields[error.form].fields[fieldName].setValidityAndMessages(
            Validity.Caution,
            [],
          );
        }
      }
    }
  });
}
