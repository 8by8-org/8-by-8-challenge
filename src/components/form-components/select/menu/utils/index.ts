export * from './focus';
export * from './keyboard';
export * from './menu-state';
export * from './scroll';
export * from './touchscreen';
