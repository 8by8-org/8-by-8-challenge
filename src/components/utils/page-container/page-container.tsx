import type { PropsWithChildren } from 'react';
import styles from './styles.module.scss';

/**
 * Renders a container that centers its content and enforces min and max widths
 * for said content.
 */
export function PageContainer({ children }: PropsWithChildren) {
  return (
    <div className={styles.outer_container}>
      <div className={styles.inner_container}>{children}</div>
    </div>
  );
}
