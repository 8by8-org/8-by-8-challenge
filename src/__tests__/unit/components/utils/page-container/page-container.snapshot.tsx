import '@testing-library/jest-dom';
import { render, cleanup } from '@testing-library/react';
import { PageContainer } from '@/components/utils/page-container';

describe('PageContainer', () => {
  afterEach(cleanup);

  it('renders the PageContainer unchanged.', () => {
    const { container } = render(<PageContainer />);
    expect(container).toMatchSnapshot();
  });

  it('applies the correct min-height style.', () => {
    const { getByTestId } = render(<PageContainer />);
    const pageContainer = getByTestId('page-container');
    const navbarHeight = '57px'; // Adjust this 
    const footerHeight = '40px'; // Adjust 
    const expectedMinHeight = `calc(100vh - ${navbarHeight} - ${footerHeight})`;

    console.log('Rendered min-height:', pageContainer.style.minHeight);

    expect(pageContainer).toHaveStyle(`min-height: ${expectedMinHeight}`);
  });
});
