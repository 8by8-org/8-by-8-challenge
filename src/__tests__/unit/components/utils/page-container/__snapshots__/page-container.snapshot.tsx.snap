// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageContainer renders the PageContainer unchanged. 1`] = `
<div>
  <div
    class="outer_container"
    data-testid="page-container"
  >
    <div
      class="inner_container"
    />
  </div>
</div>
`;
